src/main/java/roomescape/reservation/entity/Reservation.java

```diff
@@ -1,32 +1,70 @@
 package roomescape.reservation.entity;
 
-import java.time.LocalDateTime;
+import java.time.LocalDate;
+import java.util.Objects;
 
 public class Reservation {
 
-    private final long id;
+    private final Long id;
     private final String name;
-    private final LocalDateTime dateTime;
+    private final LocalDate reservationDate;
+    private final ReservationTime reservationTime;
 
-    public Reservation(long id, String name, LocalDateTime dateTime) {
+    public Reservation(Long id, String name, LocalDate reservationDate, ReservationTime reservationTime) {
+        validateReservation(name, reservationDate, reservationTime);
         this.id = id;
         this.name = name;
-        this.dateTime = dateTime;
+        this.reservationDate = reservationDate;
+        this.reservationTime = reservationTime;
     }
 
-    public boolean sameId(long id) {
-        return this.id == id;
+    public static Reservation withoutId(String name, LocalDate reservationDate, ReservationTime reservationTime) {
+        return new Reservation(null, name, reservationDate, reservationTime);
     }
 
-    public long getId() {
+    private static void validateReservation(String name, LocalDate reservationDate, ReservationTime reservationTime) {
+        if (name == null || reservationDate == null || reservationTime == null) {
+            throw new IllegalArgumentException("Reservation cannot be null");
+        }
+    }
```

Comment:
```suggestion
    public Reservation(Long id, String name, LocalDate reservationDate, ReservationTime reservationTime) {
        this.id = id;
        this.name = name;
        this.reservationDate = reservationDate;
        this.reservationTime = reservationTime;
        validateReservation();
    }

    private void validateReservation() {
        if (name == null || reservationDate == null || reservationTime == null) {
            throw new IllegalArgumentException("Reservation cannot be null");
        }
    }
```

위 방식에 대해 어떻게 생각하시는지 궁금해요~

Reply:
아 이렇게 하면 인자값을 전달할 필요가 없어져서 깔끔하네요 배워갑니다 감사합니다!

[refactor: 도메인 검증 로직 변경](https://github.com/woowacourse/spring-roomescape-admin/pull/287/commits/bf756234eb0a52a359f91e0d38821895ef87c293)